### PR TITLE
dotCMS/core#14505 Create detail page field in the content types

### PR DESCRIPTION
**Pull request by fmontes**

### CODE REVIEW

File: src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
+import { Injectable } from '@angular/core';
```
Comment (fmontes) [inline src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts]: The service needs work, trying to get the data using ElasticSearch

File: src/app/view/components/_common/dot-page-selector/dot-page-selector.component.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+     */
+    onClear(): void {
+        this.propagateChange(null);
+        this.results = [];
+    }
+
+    /**
+     * Handle option selected
+     *
+     * @param {*} $event
```
Comment (alfredo-dotcms) [inline src/app/view/components/_common/dot-page-selector/dot-page-selector.component.ts]: fix doc

File: src/app/portlets/content-types/form/content-types-form.component.spec.ts [hunk 3]
```diff
@@ -93,7 +95,8 @@ describe('ContentTypesFormComponent', () => {
                     TabViewModule,
                     SiteSelectorFieldModule,
                     RouterTestingModule,
-                    MdInputTextModule,
+                    DotDirectivesModule,
+                    DotPageSelectorModule,
                     DotWorkflowsSelectorFieldModule
                 ],
                 providers: [
```
Comment (freddyucv) [inline src/app/portlets/content-types/form/content-types-form.component.spec.ts]: We need a test to check the 'dot-page-selector'  and its attributes

File: src/app/view/components/_common/dot-page-selector/dot-page-selector.component.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        (onSelect)="onSelect($event)"
+        [inputStyle]="{'width': '100%', 'padding': '0'}"
+        [minLength]="3"
+        [ngModel]="val"
+        [placeholder]="label"
+        [style]="style"
+        [suggestions]="results"
+        field="url">
+    </p-autoComplete>
+</ng-template>
```
Comment (freddyucv) [inline src/app/view/components/_common/dot-page-selector/dot-page-selector.component.html]: All the <p-autoComplete is repeat here, I remove the floatingLabel attribute and if we need it in the future then we'll add it

File: src/app/view/components/_common/dot-page-selector/dot-page-selector.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+    it('should search for pages', () => {
+        autocomplete.triggerEventHandler('completeMethod', {
+            query: 'hello'
+        });
+
+        expect(dotPageSelectorService.getPagesInFolder).toHaveBeenCalledWith('hello');
+    });
+
+    it('should pass attritubes', () => {
```
Comment (freddyucv) [inline src/app/view/components/_common/dot-page-selector/dot-page-selector.component.spec.ts]: I think this test's description is not right

File: src/app/view/components/_common/dot-page-selector/dot-page-selector.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    constructor(private fb: FormBuilder) {
+        /*
+            This should go in the ngOnInit but I don't want to detectChanges everytime for
+            this fake test component
+        */
+        this.form = this.fb.group({
+            page: [{ value: 'f9fc55e7-557a-4047-a8be-15e5ca69fa62', disabled: false }]
+        });
+    }
+}
```
Comment (freddyucv) [inline src/app/view/components/_common/dot-page-selector/dot-page-selector.component.spec.ts]: I think is better add a attribute 'floatingLabel' for this componente and just set it to false for the rest of the test

File: src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+    folder: string;
+    __DOTNAME__: string;
+    sortOrder: number;
+    seokeywords: string;
+    modUser: string;
+    host: string;
+    lastReview: string;
+    stInode: string;
+    url: string;
+}
```
Comment (freddyucv) [inline src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts]: why is not possible use 'DotPage' here?

File: src/app/view/components/_common/dot-page-selector/dot-page-selector.component.html (hunk 4 above)
Comment (fmontes) [reply src/app/view/components/_common/dot-page-selector/dot-page-selector.component.html]: 100% disagreed, I don't want a field that is attached to a floating label effect, the floating label effect is optional, I want my fields to be, act an look as native as possible.

File: src/app/view/components/_common/dot-page-selector/dot-page-selector.component.spec.ts (hunk 5 above)
Comment (fmontes) [reply src/app/view/components/_common/dot-page-selector/dot-page-selector.component.spec.ts]: Ideas?

File: src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts (hunk 7 above)
Comment (fmontes) [reply src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts]: Too general. In fact we need to do something better with this, we need to create like `DotContentlet` and all this things should extend from that one.

File: src/app/portlets/content-types/form/content-types-form.component.spec.ts [hunk 8]
```diff
@@ -511,7 +514,7 @@ describe('ContentTypesFormComponent', () => {
                 clazz: 'com.dotcms.contenttype.model.field.ImmutableDateTimeField',
                 id: '123',
                 indexed: true,
-                name: 'publishDateVar'
+                name: 'publishDateVar',
```
Comment (hmoreras) [inline src/app/portlets/content-types/form/content-types-form.component.spec.ts]: extra semicolon

File: src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.spec.ts [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+                    body: {
+                        contentlets: [mockPageSelector]
+                    }
+                })
+            )
+        );
+        expect(result).toEqual(mockPageSelector);
+    });
+
+    it('should get a specific Page Selector', () => {
```
Comment (fmontes) [inline src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.spec.ts]: Is not getting an specific page selector, is getting a page by id.

Also need to test that the url and the body is being set to get the data is correct.

File: src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.spec.ts [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+describe('Service: DotPageSelector', () => {
+    beforeEach(() => {
+        this.injector = DOTTestBed.resolveAndCreate([DotPageSelectorService]);
+        this.dotPageSelectorService = this.injector.get(DotPageSelectorService);
+        this.backend = this.injector.get(ConnectionBackend) as MockBackend;
+        this.backend.connections.subscribe((connection: any) => (this.lastConnection = connection));
+    });
+
+    it('should get Page Selectors', () => {
```
Comment (fmontes) [inline src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.spec.ts]: Is not getting an specific page selectors, is getting a pages in a folder.

Also need to test that the url and the body is being set to get the data is correct.

File: src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+    constructor(private coreWebService: CoreWebService) {}
+
+    /**
+     * Get all the pages in the folder
+     *
+     * @param {string} _folder
+     * @returns {Observable<DotPageAsset[]>}
+     * @memberof DotPageSelectorService
+     */
+    getPagesInFolder(_folder: string): Observable<DotPageAsset[]> {
```
Comment (fmontes) [inline src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts]: why `_folder`?

**Pull request merged**